src/transformers/trainer.py

```diff
@@ -399,6 +399,9 @@ def __init__(
                     " boolean argument which will be triggered after the last batch of the eval set to signal that the"
                     " summary statistics should be returned by the function."
                 )
+        if not args.gradient_accumulation_steps or args.gradient_accumulation_steps < 1 :
+            raise ValueError(f"gradient_accumulation_steps in the TrainingArguments takes a positive integer, not {args.gradient_accumulation_steps}")
```

Comment:
This should be asserted in the training args init, rather than the trainer class

Reply:
Yes, I completely agree. To me, the concerning factor is if it is set to a negative value (by mistake, or for whatever reason), the program runs and leads to **logical error**! If you agree, please add the checking in the TrainingArguments. I leave the decision to you!

Reply:
As setting this to a negative value is very unlikely I think we should just close  the PR to add the check